### PR TITLE
Datetime sinusoidal normalizer

### PR DESCRIPTION
**Pull request by paxcema**
~~Added a class to transform a datetime() object into a tensor of sinusoidal descriptors. Integration to the actual TS RNN encoder will be done after the joint column encoding feature is deployed.~~

Adds capability of encoding each date/time stamp using sinusoidal functions to the `DatetimeEncoder` class.

This PR also fixes a small bug in the text RNN decoding step and a `batch_size` argument for the hidden state initializer.

### CODE REVIEW

File: lightwood/encoders/time_series/helpers/rnn_helpers.py [hunk 1]
```diff
@@ -93,3 +97,45 @@ def fit_transform(self, x):
 
     def inverse_transform(self, y):
         return self.scaler.inverse_transform(y)
+
+
+class DateNormalizer:
```
Comment (George3d6) [inline lightwood/encoders/time_series/helpers/rnn_helpers.py:102]: Would it not be possible to share this with the current date encoder ?

Comment (paxcema) [reply lightwood/encoders/time_series/helpers/rnn_helpers.py:102]: Do you mean as in adding these functionalities to the encoder class instead?

Comment (paxcema) [reply lightwood/encoders/time_series/helpers/rnn_helpers.py]: Okay, functionality is migrated into the date encoder, looks better now.

**Review by George3d6 — APPROVED** [review]

**Comment by George3d6** [comment]
By the way, this closes #265
